src/components/user/services/user.service.v2.integration.test.js

```diff
@@ -0,0 +1,163 @@
+			const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+			const user = await testObjects.createTestUser({ roles: ['student'], schoolId });
+			const token = await testObjects.generateJWTFromUser(teacher);
+			const request = chai
+				.request(app)
+				.delete(`/users/v2/admin/student/${user._id.toString()}`)
+				.query({ userId: user._id.toString() })
+				.set('Accept', 'application/json')
+				.set('Authorization', token)
+				.set('Content-type', 'application/json');
+			const response = await request.send();
+			expect(response.status).to.equal(403);
+		});
+
+		it('Returns error is user not found', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const notFoundId = ObjectId();
+			const token = await getAdminToken(schoolId);
+			const request = chai
+				.request(app)
+				.delete(`/users/v2/admin/student/${notFoundId.toString()}`)
+				.query({ userId: notFoundId.toString() })
+				.set('Accept', 'application/json')
+				.set('Authorization', token)
+				.set('Content-type', 'application/json');
+			const response = await request.send();
+			expect(response.status).to.equal(404);
+		});
+
+		it('Fails when user from different deletes a student', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const { _id: otherSchoolId } = await testObjects.createTestSchool();
+			const user = await testObjects.createTestUser({ roles: ['student'], schoolId });
+			const token = await getAdminToken(otherSchoolId);
+			const request = chai
+				.request(app)
+				.delete(`/users/v2/admin/student/${user._id.toString()}`)
+				.query({ userId: user._id.toString() })
+				.set('Accept', 'application/json')
+				.set('Authorization', token)
+				.set('Content-type', 'application/json');
+			const response = await request.send();
+			expect(response.status).to.equal(403);
+		});
+
+		it('users with STUDENT_DELETE permission can REMOVE students', async () => {
+			const school = await testObjects.createTestSchool({
+				name: 'testSchool',
+			});
+			const { _id: schoolId } = school;
+
+			const token = await getPermissionToken(schoolId, ['STUDENT_DELETE']);
+
+			const deleteUser = await testObjects.createTestUser({ roles: ['student'], schoolId });
+
+			const request = chai
+				.request(app)
+				.delete(`/users/v2/admin/student/${deleteUser._id.toString()}`)
+				.set('Accept', 'application/json')
+				.set('Authorization', token)
+				.set('Content-type', 'application/json');
+			const response = await request.send();
+			expect(response.status).to.equal(200);
+
+			const checkUser = await usersModelService.get(deleteUser._id);
+			expect(checkUser.fullName).to.equal('DELETED USER');
+
+			const checkAccount = await accountModelService.find({ query: { userId: deleteUser._id }, paginate: false });
+			expect(checkAccount.length).to.equal(0);
+		});
+
+		it('users with STUDENT_DELETE permission can not REMOVE teachers', async () => {
```

Comment:
the requirement actually is that you cant delete a teacher without TEACHER_DELETE. a user with STUDENT_DELETE would be allowed to delete if he also has that other permission.

This test title would infer that a user with STUDENT_DELETE will never be allowed to remove a teacher....